benchmark/torch/NeurIPS2020-Learning-to-Run-a-Power-Network-Challenge/track1/rl_agent.py

```diff
@@ -0,0 +1,517 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import numpy as np
+import pickle
+import os
+from grid2op.Agent import BaseAgent
+from copy import deepcopy
+from powernet_model import CombinedActionModel, UnitaryActionModel
+from utils import FeatureProcessor, UnitaryFeatureProcessor
+
+
+class RLAgent(BaseAgent):
+    def __init__(self, action_space):
+        BaseAgent.__init__(self, action_space=action_space)
+        self.simulate_times = 0
+
+        self.combined_model_1 = CombinedActionModel()
+        self.combined_model_2 = CombinedActionModel()
+        self.unitary_model = UnitaryActionModel()
+
+        unitary_actions_vec = np.load(
+            os.path.join('saved_files',
+                         "v6_top500_unitary_actions.npz"))["actions"]
+        self.unitary_actions = []
+        for i in range(unitary_actions_vec.shape[0]):
+            action = action_space.from_vect(unitary_actions_vec[i])
+            self.unitary_actions.append(action)
+
+        redispatch_actions_vec = np.load(
+            os.path.join('saved_files', "redispatch_actions.npz"))["actions"]
+        self.redispatch_actions = []
+        for i in range(redispatch_actions_vec.shape[0]):
+            action = action_space.from_vect(redispatch_actions_vec[i])
+            self.redispatch_actions.append(action)
+
+        with open(
+                os.path.join('saved_files', "action_to_sub_id.pickle"),
+                "rb") as f:
+            self.action_to_sub_id = pickle.load(f)
+
+        self.after_line56_disconnect_actions = []
```

Comment:
rename `after_line56_disconnect_actions` to `after_line56_or_line45_disconnect_actions`.
rename it in the paddle version at the same time.

Reply:
updated